service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseExtensionController.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.service.CaseExtensionService;
+import uk.gov.hmcts.reform.fpl.service.ValidateGroupService;
+import uk.gov.hmcts.reform.fpl.validation.groups.CaseExtensionGroup;
+
+import java.time.LocalDate;
+
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.DATE;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateToString;
+
+
+@Api
+@RestController
+@RequestMapping("/callback/case-extension")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CaseExtensionController {
+    private final ObjectMapper mapper;
+    private final ValidateGroupService validateGroupService;
+    private final CaseExtensionService caseExtensionService;
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        LocalDate caseCompletionDate = caseExtensionService.getCaseCompletionOrSubmittedDate(caseData);
```

Comment:
Can we return already formatted date string from this method, so we don't have to do it in controller?

Reply:
discussed